Lib/idlelib/idle_test/test_formatregion.py

```diff
@@ -0,0 +1,236 @@
+"Test formatregion, coverage 100%."
+
+from idlelib import formatregion
+import unittest
+from unittest import mock
+from test.support import requires
+from tkinter import Tk, Text
+from idlelib.editor import EditorWindow
+
+
+code_sample = """\
+
+class C1():
+    # Class comment.
+    def __init__(self, a, b):
+        self.a = a
+        self.b = b
+
+    def compare(self):
+        if a > b:
+            return a
+        elif a < b:
+            return b
+        else:
+            return None
+"""
+
+
+class DummyEditwin:
+    def __init__(self, root, text):
+        self.root = root
+        self.text = text
+        self.indentwidth = 4
+        self.tabwidth = 4
+        self.usetabs = False
+        self.context_use_ps1 = True
+
+    _make_blanks = EditorWindow._make_blanks
+    get_selection_indices = EditorWindow.get_selection_indices
+
+
+class HelperTest(unittest.TestCase):
+
+    def test_classifyws(self):
+        classify = formatregion.classifyws
+        eq = self.assertEqual
+
+        tabwidth = 4
+        tests = (('no spaces', (0, 0)),
+                 # Internal space isn't counted.
+                 ('    space test', (4, 4)),
+                 ('\ttab test', (1, 4)),
+                 ('\t\tdouble tabs test', (2, 8)),
+                 # Different results when mixing tabs and spaces.
+                 ('    \tmixed test', (5, 8)),
+                 ('  \t  mixed test', (5, 6)),
+                 ('\t    mixed test', (5, 8)),
+                 # Spaces not divisible by tabwidth.
+                 ('  \tmixed test', (3, 4)),
+                 (' \t mixed test', (3, 5)),
+                 ('\t  mixed test', (3, 6)),
+                 # Only checks spaces and tabs.
+                 ('\nnewline test', (0, 0)))
+
+        for string, expected in tests:
+            with self.subTest(string=string):
+                eq(classify(string, tabwidth), expected)
+
+
+class FormatRegionTest(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        requires('gui')
+        cls.root = Tk()
+        cls.root.withdraw()
+        cls.text = Text(cls.root)
+        cls.text.undo_block_start = mock.Mock()
+        cls.text.undo_block_stop = mock.Mock()
+        cls.editor = DummyEditwin(cls.root, cls.text)
+        cls.formatter = formatregion.FormatRegion(cls.editor)
+
+    @classmethod
+    def tearDownClass(cls):
+        del cls.text, cls.formatter, cls.editor
+        cls.root.update_idletasks()
+        cls.root.destroy()
+        del cls.root
+
+    def setUp(self):
+        self.text.insert('1.0', code_sample)
+
+    def tearDown(self):
+        self.text.delete('1.0', 'end')
+
+    def test_get_region(self):
+        get = self.formatter.get_region
```

Comment:
I find all of these assignments make the tests harder to read rather than easier. It increases the number of things I need to remember, and makes it necessary to read every test method from the beginning to understand any part of it.

This is a matter of style, but IMO when using many short test methods as done here (which I like!) these "shortcuts" do more harm than good.

To be clear, this is true for all of the test methods here.

Reply:
Looking more, all functions start with func =, text =, eq =, and I think uniformity is better than variety here.